odoo/addons/base/models/ir_mail_server.py

```diff
@@ -143,6 +143,46 @@ def _check_smtp_ssl_files(self):
             elif mail_server.smtp_ssl_private_key and not mail_server.smtp_ssl_certificate:
                 raise UserError(_('SSL certificate is missing for %s.', mail_server.name))
 
+    def write(self, vals):
+        """Ensure we cannot archive a server in-use"""
+        has_archive_error = False
+        if not vals.get('active', True):
+            usages = self._active_usages_compute()
+            if usages:
+                has_archive_error = True
+
+        if not has_archive_error:
+            return super().write(vals)
+
+        usages = {s: u for s, u in usages.items() if u}
```

Comment:
Could you use meaningful variable names ?

Reply:
Yes, I have updated the whole method following your next comment.